test/common/report.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+
+const REPORT_SECTIONS = [ 'header',
+                          'javascriptStack',
+                          'nativeStack',
+                          'javascriptHeap',
+                          'resourceUsage',
+                          'libuv',
+                          'environmentVariables',
+                          'userLimits',
+                          'sharedObjects' ];
+
+let tmppath = '';
+
+exports.findReports = (pid, path) => {
+  // Default filenames are of the form
+  // node-report.<date>.<time>.<pid>.<seq>.json
+  tmppath = path;
+  const format = '^node-report\\.\\d+\\.\\d+\\.' + pid + '\\.\\d+\\.json$';
+  const filePattern = new RegExp(format);
+  const files = fs.readdirSync(path);
+  return files.filter((file) => filePattern.test(file));
+};
+
+exports.validate = (report, options) => {
+  const jtmp = path.join(tmppath, report);
+  fs.readFile(jtmp, (err, data) => {
+    this.validateContent(data, options);
+  });
+};
+
+
+exports.validateContent = function validateContent(data, options) {
+  const report = JSON.parse(data);
+  const comp = Object.keys(report);
+
+  // Check all sections are present
+  REPORT_SECTIONS.forEach((section) => {
+    assert.ok(comp.includes(section));
+  });
+
+/*  process.on('exit', () => {
```

Comment:
Sorry if I missed the explanation, but why is this commented out?

Reply:
this was in place when we designed the tests to run sequentially and in the process running in the same folder - to cleanup the generated reports. Now we changed those to run in a temp folder that gets cleared by the test harness. So this block is not required anymore, I removed it.